lib/python/ray/worker.py

```diff
@@ -511,13 +610,17 @@ def process_task(task): # wrapping these lines in a function should cause the lo
     else:
       store_outputs_in_objstore(return_objrefs, outputs, worker) # store output in local object store
       ray.lib.notify_task_completed(worker.handle, True, "") # notify the scheduler that the task completed successfully
+    finally:
+      # Reinitialize the values of reusable variables that were used in the task
+      # above so that changes made to their state do not affect other tasks.
+      ray.reusables._reinitialize()
   while True:
-    (task, function) = ray.lib.wait_for_next_message(worker.handle)
+    (task, function, variable_info) = ray.lib.wait_for_next_message(worker.handle)
     try:
       # Currently the schedule does not ask the worker to execute a task and
       # import a function at the same time.
-      assert task is None or function is None
-      if task is None and function is None:
+      assert (task is not None) + (function is not None) + (variable_info is not None) <= 1
```

Comment:
Factor out common code, e.g.

	sum(map(lambda obj: obj is not None, (task, function, variable_info))) <= 1

(my preference) or

	sum(obj is not None for obj in (task, function, variable_info)) <= 1

(more popular preference).